src/app/main.cpp

```diff
@@ -862,11 +863,29 @@ int main( int argc, char *argv[] )
   // Settings migration is only supported on the default profile for now.
   if ( profileName == "default" )
   {
+    // Note: this flag is ka version number so that we can reset it once we change the version.
+    // Note2: Is this a good idea can we do it better.
+
+    int firstRunVersion = settings.value( QStringLiteral( "migration/firstRunVersionFlag" ), 0 ).toInt();
+    bool showWelcome = ( firstRunVersion == 0 || firstRunVersion != Qgis::QGIS_VERSION_INT );
```

Comment:
@m-kuhn how does this feel to you? I want a way to trigger this on release again. But this will also trigger again when we bump past 3000 so might need a better way. Having said that it will only show if you haven't already run the migration so I guess that is ok. e.g this will only run if we need to do migration as well as this pass so I think it will be ok.

Reply:
`|| Qgis::QGIS_VERSION_INT > firstRunVersion`?

Reply:
Yeah although I guess that is really the same result.  I think this kind of check is ok like I said it will only get run if the version mismatches and also the settings haven't been migrated.